specification/snapshot/_types/SnapshotRepository.ts

```diff
@@ -17,22 +17,126 @@
  * under the License.
  */
 
-import { Uuid } from '@_types/common'
+import { ByteSize, Uuid } from '@_types/common'
 import { integer } from '@_types/Numeric'
+import { Duration } from '@_types/Time'
 
-export class Repository {
-  type: string
+export type Repository =
```

Comment:
We need [a variant annotation indicating that `type` is the discriminator](https://github.com/elastic/elasticsearch-specification/blob/main/docs/modeling-guide.md#variants), and also the indication that the enumeration isn't exhaustive as new implementations can (theoretically) be added using plugins.

```ts
/**
 * @variants internal tag='type'
 * @non_exhaustive
 */
```

Reply:
Fixed in ac703f9845694604552925c12332b73ceaf8cc85